uploadNeuroDB/imaging_non_minc_insertion.pl

```diff
@@ -0,0 +1,574 @@
+    $message = "\n\tERROR: Invalid UploadID $upload_id.\n\n";
+    # write error message in the log file
+    $utility->writeErrorLog(
+        $message, $NeuroDB::ExitCodes::SELECT_FAILURE, $log_file
+    );
+    # insert error message into notification spool table
+    $notifier->spool(
+        'imaging non minc file insertion'   , $message,   0,
+        'imaging_non_minc_insertion.pl', $upload_id, 'Y',
+        'N'
+    );
+    exit $NeuroDB::ExitCodes::SELECT_FAILURE;
+}
+
+
+
+
+###### Determine the acquisition protocol ID of the file based on scan type
+
+# verify that an acquisition protocol ID exists for $scan_type
+my $acqProtocolID = NeuroDB::MRI::scan_type_text_to_id($scan_type, \$dbh);
+if ($acqProtocolID =~ /unknown/){
+    $message = "\n\tERROR: no AcquisitionProtocolID found for $scan_type.\n\n";
+    # write error message in the log file
+    $utility->writeErrorLog( $message, $NeuroDB::ExitCodes::UNKNOWN_PROTOCOL, $log_file );
+    # insert error message into notification spool table
+    $notifier->spool(
+        'imaging non minc file insertion',    $message,   0,
+        'imaging_non_minc_insertion.pl', $upload_id, 'Y',
+        'N'
+    );
+    exit $NeuroDB::ExitCodes::UNKNOWN_PROTOCOL;
+} else {
+    $message = "\nFound protocol ID $acqProtocolID for $scan_type.\n\n";
+    $notifier->spool(
+        'imaging non minc file insertion',    $message,   0,
+        'imaging_non_minc_insertion.pl', $upload_id, 'Y',
+        'Y'
+    )
+}
+
+
+
+
+###### Create and Load File object
+
+# create File object
+my $file = NeuroDB::File->new(\$dbh);
+
+# file type determined here as long as the it exists in ImagingFileTypes table
+$file->loadFileFromDisk($file_path);
+
+
+
+
+###### Determine file basename and directory name
+
+# Get the file name and directory name of $file_path
+my ($file_name, $dir_name) = fileparse($file_path);
+
+
+
+
+###### Determine the metadata to be stored in parameter_file if a JSON metadata
+###### file is provided
+
+if ($metadata_file) {
+    # read the JSON file into $json
+    local $/;  # Enable 'slurp' mode
+    open( FILE, $metadata_file ) or die "Could not open file: $!";
+    my $json = <FILE>;
+    close( FILE );
+
+    my $metadata = decode_json($json);
+    foreach my $parameter (sort keys $metadata) {
+        $file->setParameter( $parameter, $metadata->{$parameter} );
+    }
+}
+
+
+
+
+###### Determine candidate information
+
+# create a hash similar to tarchiveInfo so that can use Utility routines to
+# determine the candidate information
+# if patient name provided as an argument, use it to get candidate/site info
+# otherwise, use the file's name to determine candidate & site information
+my %info = (
+    SourceLocation => undef,
+    PatientName    => ($patient_name // $file_name),
+    PatientID      => ($patient_name // $file_name)
+);
+
+# determine subject ID information
+my $subjectIDsref = $utility->determineSubjectID($scanner_id, \%info, 0);
+unless ($subjectIDsref){
```

Comment:
I think what you really want here is 
```perl
unless (%$subjectIDsref) {
```
otherwise the `unless` code will never fire since `$subjectIDsref` is always `defined`.

Reply:
done